src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -332,12 +321,10 @@ private void check() {
                             recurrencePeriod = 0;
                             controller.cleanup(workspace);
                         }
-                        return null;
-                    }
-                }, 10, TimeUnit.SECONDS);
             } catch (Exception x) {
                 LOGGER.log(Level.FINE, "could not check " + workspace, x);
                 ws = null;
+                logger().println("Cannot contact " + node + ": " + x); // TODO as above
```

Comment:
Hi,

We have an issue in our Jenkins: in a pipeline job, a part of a job started on a docker template will never finish if the the machine running the container goes offline. The build remains stuck on this log:
Cannot contact swarm-41aec2277655: hudson.remoting.RequestAbortedException: java.nio.channels.ClosedChannelException

My question is : why there is no exception raised here to stop the build and allow to implement a retry strategy?

Any help would be very appreciated.

Thanks.